lib/xhttp.ex

```diff
@@ -86,32 +312,223 @@ defmodule XHTTP do
   def request(conn, method, path, headers, body \\ nil),
     do: conn_module(conn).request(conn, method, path, headers, body)
 
+  @doc """
+  Streams a chunk of the request body on the connection or signals the end of the body.
+
+  If a request is opened (through `request/5`) with the body as `:stream`, then the
+  body can be streamed through this function. The function takes a `conn`, a
+  `request_ref` returned by `request/5` to identify the request to stream the body for,
+  and a chunk of body to stream. The value of chunk can be:
+
+    * iodata - a chunk of iodata is transmitted to the server as part of the body
+      of the request.
+
+    * `:eof` - signals the end of the streaming of the request body for the given
+      request. Usually the server won't send any reply until this is sent.
+
+  This function always returns an updated connection to be stored over the old connection.
+
+  ## Examples
+
+  Let's see an example of streaming an empty JSON object (`{}`) by streaming one curly
+  brace at a time.
+
+      headers = [{"Content-Type", "application/json"}, {"Content-Length", "2"}]
```

Comment:
```suggestion
      headers = [{"content-type", "application/json"}, {"content-length", "2"}]
```

Lower case headers again.